src/content/tutorials/fleet-management/deletion-prevention/index.md

```diff
@@ -0,0 +1,153 @@
+---
+linkTitle: Prevent deletion of resources
+title: Prevent accidental deletion of resources
+description: Avoid accidental deletion of clusters, apps or other resources in Giant Swarm platform.
+weight: 110
+menu:
+  main:
+    parent: advanced-app-platform
+user_questions:
+  - How can I prevent the accidental deletion of resources?
+  - How can I protect resources from accidental deletion?
+  - How can I safeguard resources?
+  - How can I protect clusters from accidental deletion?
+  - How can I safeguard clusters?
+last_review_date: 2023-11-07
+aliases:
+  - /advanced/app-platform/deletion-prevention
+  - /guides/deletion-prevention/
+  - /advanced/deletion-prevention/
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-tenet
+---
+
+
+In Kubernetes environments, it can be quite easy to accidentally delete an important object with `kubectl delete`. To mitigate such risks, Giant Swarm clusters have Kyverno policy which safeguards against deletions of resources with the `giantswarm.io/prevent-deletion` label.
+
+## How the label works
+
+To use this mechanism you have to do two things:
+
+1. Ensure that the resource type of the object you want to protect is in the [list of targeted resources](https://github.com/giantswarm/kyverno-policies-ux/blob/main/helm/kyverno-policies-ux/values.yaml). You can also show the preinstalled policy to see the list:
+
+   ```nohighlight
+   $ kubectl get ClusterPolicy/block-resource-deletion-if-has-prevent-deletion-label -o yaml
+   [...]
+     rules:
+     - match:
+         any:
+         - resources:
+             kinds:
+             - Cluster
+             - Cluster
+             - Cluster
+             - Cluster
+             - Cluster
+             - MachineDeployment
+             - MachinePool
+             - AWSCluster
+             - App
+             - Organization
+             - Namespace
+             - Secret
+             - Secret
+             - ConfigMap
+             - KubeadmControlPlane
+             selector:
+               matchLabels:
+                 giantswarm.io/prevent-deletion: '*'
+   [...]
+   ```
+
+   If you want to extend this list of object types, please continue reading below about how to configure it.
+2. Add the `giantswarm.io/prevent-deletion: "true"` label to the desired object (the `"true"` value does not matter). Below, we explain how to easily do this for clusters or managed apps.
+
+## Creating a cluster with deletion prevention
+
+In the `kubectl gs template cluster` command of the [Create a workload cluster]({{< relref "/getting-started/provision-your-first-workload-cluster/" >}}) guide, add the parameter `--prevent-deletion`.
+
+## Adding deletion prevention to an existing cluster
+
+Our CAPI-based cluster charts allow setting the option `global.metadata.preventDeletion=true`. It can be toggled in the values of the chart:
```

Comment:
```suggestion
Our Cluster API (CAPI) based cluster charts allow setting the option `global.metadata.preventDeletion=true`. It can be toggled in the values of the chart:
```